### PR TITLE
Move the authentication middleware to a new OrchardCore.Authentication module

### PR DESCRIPTION
**Pull request by PinpointTownes**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/1392.

/cc @sebastienros

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Users/Module.txt [hunk 1]
```diff
@@ -5,4 +5,5 @@ Website: http://orchardproject.net
 Version: 2.0.x
 OrchardVersion: 2.0.x
 Description: The users module enables authentication UI and user management.
+Dependencies: OrchardCore.Authentication
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/OrchardCore.Users/Module.txt]: @sebastienros should the `OrchardCore.Users` package also have a (NuGet) dependency on `OrchardCore.Authentication`?

Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.Users/Module.txt]: It's not a physical dependency so no. Even though this feature won't work without the other feature enabled. This just ensure that if one day we have another feature in the manifest that doesn't require it, then it won't download it for nothing.

File: src/OrchardCore.Modules/OrchardCore.Authentication/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            app.UseAuthentication();
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddAuthentication();
+
+            // Note: IAuthenticationSchemeProvider is already registered at the host level.
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Authentication/Startup.cs]: Is it making the host one useless then? Do we need it in any other scenario?

File: src/OrchardCore.Modules/OrchardCore.Users/Module.txt (hunk 1 above)
Comment (PinpointTownes) [reply src/OrchardCore.Modules/OrchardCore.Users/Module.txt]: Well, if we don't transitively reference the authentication module, developers will have to do it themselves before being able to use the users module. If that's not a problem, that's fine.

File: src/OrchardCore.Modules/OrchardCore.Authentication/Startup.cs (hunk 2 above)
Comment (PinpointTownes) [reply src/OrchardCore.Modules/OrchardCore.Authentication/Startup.cs]: I suppose developers could still register non-tenant aware handlers at the host level.

Note that this code is not new. It was just moved from the Users module.

**Pull request merged**